osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationServiceTest.java

```diff
@@ -51,10 +51,10 @@ void addEMeter() throws FunctionalException {
         when(this.dlmsDeviceRepository.save(dlmsDevice)).thenReturn(dlmsDevice);
         when(this.encryptionHelperService.rsaDecrypt(any())).thenReturn(new byte[16]);
         // WHEN
-        this.testService.addMeter(deviceDto);
-        //THEN
-        verify(this.secretManagementService, times(1)).storeNewKeys(any(), any());
-        verify(this.secretManagementService, times(1)).activateNewKeys(any(), any());
+        this.testService.addMeter("test", deviceDto);
+        // THEN
+        verify(this.secretManagementService, times(1)).storeNewKeys(any(), any(), any());
+        verify(this.secretManagementService, times(1)).activateNewKeys(any(), any(), any());
```

Comment:
Would it be a good idea to put the `correlationUid` and `deviceIdentification` in variables, and call `verify` like:
```java
verify(this.secretManagementService, times(1)).storeNewKeys(eq(correlationUid), eq(deviceIdentification), any());
verify(this.secretManagementService, times(1)).activateNewKeys(eq(correlationUid), eq(deviceIdentification), any());
```
The installation service should pass on the appropriate values for the request flow that is tested, not random strings. Perhaps even the keys should be checked more thoroughly, although that would require more work than checking the `String` values.

(Something similar would go for the other tests where the correlation UID was added in this and a few other test classes modified in this pull request.)

Reply:
Now testing parameter equals (eq().
CorrelationUid parameter is now MessageMetadata